### PR TITLE
License functionality

### PR DESCRIPTION
**Pull request by andrewlimaza**
* License functionality.

* TODO: Add information about SWS/Links on the relevant pages.

### CODE REVIEW

File: includes/classes/class-swsales-license.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+					<a class="button button-hero" href="https://www.paidmembershipspro.com/pricing/?utm_source=plugin&utm_medium=pmpro-license&utm_campaign=pricing&utm_content=view-license-options" target="_blank"><?php echo esc_html( 'View Support License Options', 'paid-memberships-pro' ); ?></a>
+				<?php } else { ?>
+					<a class="button button-primary button-hero" href="https://www.paidmembershipspro.com/membership-account/?redirect_to=%2Fmembership-account%2F%3Futm_source%3Dplugin%26utm_medium%3Dpmpro-license%26utm_campaign%3Dmembership-account%26utm_content%3Dview-account" target="_blank"><?php echo esc_html( 'Manage My Account', 'paid-memberships-pro' ); ?></a>
+					<a class="button button-hero" href="https://www.paidmembershipspro.com/membership-account/?redirect_to=%2Fnew-topic%2F%3Futm_source%3Dplugin%26utm_medium%3Dpmpro-license%26utm_campaign%3Dsupport%26utm_content%3Dnew-support-ticket" target="_blank"><?php echo esc_html( 'Open Support Ticket', 'paid-memberships-pro' ); ?></a>
```
Comment (dparker1005) [inline includes/classes/class-swsales-license.php:95]: Should be updated to use sitewide-sales text domain and link to Strangers Studios pages

File: includes/classes/class-swsales-license.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+            <?php
+                echo '<p>' . wp_kses( __( '<strong>Sitewide Sales offers plans for automatic updates and premium support.</strong> These plans include a Plus license key which we recommend for all public websites running Sitewide Sales. A Plus license key allows you to automatically update when a new security, bug fix, or feature enhancement is released.' ), $allowed_pmpro_license_strings_html ) . '</p>';
+            ?>
+
+            <?php
+                echo '<p>' . wp_kses( __( '<strong>Need help?</strong> Your license allows you to open new tickets in our private support area. Purchases are backed by a 30 day, no questions asked refund policy.' ), $allowed_pmpro_license_strings_html ) . '</p>';
+            ?>
+
+            <p><a href="#" target="_blank"><?php echo esc_html( 'View Support License Options &raquo;', 'paid-memberships-pro' ); ?></a></p>
```
Comment (dparker1005) [inline includes/classes/class-swsales-license.php:127]: Should be updated to use sitewide-sales text domain and link to Stranger Studios support, and may want to change var name `$allowed_pmpro_license_strings_html`. Also text domain on 116

File: includes/classes/class-swsales-license.php [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+			'sitewide_sales_license',
+			array( __CLASS__, 'show_license_page' )
+		);
+    }
+
+    public static function show_license_page() {
+
+        //only let admins get here
+        if ( ! function_exists( 'current_user_can' ) || ( ! current_user_can( 'manage_options' ) && ! current_user_can( 'sws_license') ) ) {
+            die( __( 'You do not have permissions to perform this action.', 'paid-memberships-pro' ) );
```
Comment (dparker1005) [inline includes/classes/class-swsales-license.php:28]: Should use sitewide-sales text domain

File: includes/classes/class-swsales-license.php [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                <?php } elseif(!sws_license_is_valid()) { ?>
+                    <p class="sitewide_sales_message sitewide_sales_error"><strong><?php _e('Your license is invalid or expired.', 'paid-memberships-pro' );?></strong> <?php _e('Visit the PMPro <a href="https://www.paidmembershipspro.com/membership-account/?redirect_to=%2Fmembership-account%2F%3Futm_source%3Dplugin%26utm_medium%3Dpmpro-license%26utm_campaign%3Dmembership-account%26utm_content%3Dkey-not-valid" target="_blank">Membership Account</a> page to confirm that your account is active and to find your license key.', 'paid-memberships-pro' );?></p>
+                <?php } else { ?>													
+                    <p class="sitewide_sales_message sitewide_sales_success"><?php printf(__('<strong>Thank you!</strong> A valid <strong>%s</strong> license key has been used to activate your support license on this site.', 'paid-memberships-pro' ), ucwords($sws_license_check['license']));?></p>
```
Comment (dparker1005) [inline includes/classes/class-swsales-license.php:71]: Should be updated to use sitewide-sales text domain and link to Strangers Studios pages

File: includes/license.php [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    if ( current_time( 'timestamp' ) < $sws_nag_paused && $sws_nag_paused < current_time( 'timestamp' ) * 3600*24*35 ) {
+        return;
+    }
+
+    // Get the key to use later.
+    $key = get_option( 'sws_license_key' );
+
+    //okay, show nag
+    ?>
+    <div class="<?php if(!empty($key)) { ?>sitewide_sales_message sitewide_sales_error<?php } else { ?>sitewide_sales_message sitewide_sales_error<?php } ?> fade">
```
Comment (dparker1005) [inline includes/license.php:166]: Both cases of `if` statement see to do the same thing

File: includes/license.php [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+}
+register_activation_hook( __FILE__, 'sws_license_activation' );
+
+/**
+ * Remove monthly check when plugin is deactivated.
+ */
+function sws_license_deactivation() {
+	wp_clear_scheduled_hook( 'sws_license_check_key' );
+}
+register_deactivation_hook( __FILE__, 'pmpro_deactivation' );
```
Comment (dparker1005) [inline includes/license.php:21]: Need to hook `'sws_license_deactivation'`

File: includes/classes/class-swsales-license.php (hunk 1 above)
Comment (andrewlimaza) [reply includes/classes/class-swsales-license.php:95]: These URL's also need to be updated with latest SS site links. @ideadude

File: includes/license.php (hunk 6 above)
Comment (andrewlimaza) [reply includes/license.php:21]: Doh! Thanks!

Comment (andrewlimaza) [reply includes/license.php]: This has been adjusted.

File: includes/classes/class-swsales-license.php (hunk 2 above)
Comment (andrewlimaza) [reply includes/classes/class-swsales-license.php]: This has been updated.

File: includes/classes/class-swsales-license.php (hunk 3 above)
Comment (andrewlimaza) [reply includes/classes/class-swsales-license.php]: This has been updated.

File: includes/classes/class-swsales-license.php (hunk 4 above)
Comment (andrewlimaza) [reply includes/classes/class-swsales-license.php]: This has been updated.

File: includes/license.php (hunk 5 above)
Comment (andrewlimaza) [reply includes/license.php]: This has been updated.

File: includes/license.php (hunk 6 above)
Comment (andrewlimaza) [reply includes/license.php]: This has been updated.

**Comment by andrewlimaza** [comment]
I've adjusted most of these changes, all that is left to do is link to the relevant URL's of the new site.

File: includes/classes/class-swsales-license.php (hunk 1 above)
Comment (ideadude) [reply includes/classes/class-swsales-license.php]: I'll merge in then update.

**Pull request merged**